app/cdap/components/NamespaceAdmin/SourceControlManagement/SourceControlManagementModal.tsx

```diff
@@ -0,0 +1,291 @@
+            onChange={(e) => {
+              handleValueChange(e, 'SET_REPOSITORY_URL');
+              setLinkError(!e.target.value);
+            }}
+            required
+            error={linkError}
+          />
+          <StyledTextField
+            label={T.translate(`${PREFIX}.branch.label`)}
+            value={formState?.defaultBranch}
+            helperText={T.translate(`${PREFIX}.branch.helperText`)}
+            onChange={(e) => {
+              handleValueChange(e, 'SET_BRANCH');
+            }}
+          />
+          <StyledTextField
+            label={T.translate(`${PREFIX}.pathPrefix.label`)}
+            value={formState?.pathPrefix}
+            helperText={T.translate(`${PREFIX}.pathPrefix.helperText`)}
+            onChange={(e) => {
+              handleValueChange(e, 'SET_PATH_PREFIX');
+            }}
+          />
+        </FormGroup>
+        <BoldHeader>{T.translate(`${PREFIX}.authHeader`)}</BoldHeader>
+        <FormLabel>{T.translate(`${PREFIX}.auth.label`)}</FormLabel>
+        <RadioGroup
+          row
+          value={formState?.auth?.type ? formState.auth.type : scmAuthType.pat}
+          onChange={(e) => {
+            handleValueChange(e, 'SET_AUTH_TYPE');
+          }}
+        >
+          <FormControlLabel
+            value={scmAuthType.pat}
+            control={<Radio />}
+            label={T.translate(`${PREFIX}.auth.pat.label`)}
+          />
+        </RadioGroup>
+        {formState?.auth?.type === scmAuthType.pat && (
+          <FormGroup>
+            <StyledTextField
+              label={T.translate(`${PREFIX}.auth.pat.tokenName`)}
+              value={formState?.auth?.tokenName}
+              helperText={T.translate(`${PREFIX}.auth.pat.tokenNameHelperText`)}
+              onChange={(e) => {
+                handleValueChange(e, 'SET_AUTH_TOKEN_NAME');
+                setTokenNameError(!e.target.value);
+              }}
+              required
+              error={tokenNameError}
+            />
+            <StyledTextField
+              label={T.translate(`${PREFIX}.auth.pat.token`)}
+              value={formState?.auth?.token}
+              helperText={T.translate(`${PREFIX}.auth.pat.tokenHelperText`)}
+              onChange={(e) => {
+                handleValueChange(e, 'SET_AUTH_TOKEN');
+                setTokenError(!e.target.value);
+              }}
+              required
+              error={tokenError}
+            />
+            <StyledTextField
+              label={T.translate(`${PREFIX}.auth.pat.username`)}
+              value={formState?.auth?.username}
+              helperText={T.translate(`${PREFIX}.auth.pat.usernameHelperText`)}
+              onChange={(e) => {
+                handleValueChange(e, 'SET_AUTH_USER_NAME');
+              }}
+            />
+          </FormGroup>
+        )}
+        <PrimaryContainedButton
+          children={T.translate(`${PREFIX}.validate.button`)}
```

Comment:
why are you doing children= on these instead of just putting the child wrapped with the button

Reply:
addressed all the children